### PR TITLE
Use cache_dir instead of local_dir

### PR DESCRIPTION
**Pull request by jordimas**
See https://github.com/guillaumekln/faster-whisper/issues/181

### CODE REVIEW

**Comment by jordimas** [comment]
Sure, I will do in the next hour.

**Comment by jordimas** [comment]
@guillaumekln I have added the new param. Thanks

File: faster_whisper/transcribe.py [hunk 1]
```diff
@@ -105,14 +106,15 @@ def __init__(
...
+          cache_dir: Path to the folder where cached files are stored.
         """
         self.logger = get_logger()
 
         if os.path.isdir(model_size_or_path):
             model_path = model_size_or_path
         else:
             model_path = download_model(
-                model_size_or_path, download_root, local_files_only
+                model_size_or_path, download_root, local_files_only, cache_dir
```
Comment (guillaumekln) [inline faster_whisper/transcribe.py:117]: It seems that in openai/whisper, `download_root` sets the cache directory and not the destination model path. So we could just pass `download_root` to the new `cache_dir` argument.

```suggestion
                model_size_or_path,
                local_files_only=local_files_only,
                cache_dir=download_root,
```

What do you think?

Comment (guillaumekln) [reply faster_whisper/transcribe.py:117]: No, the latest version of `download_model` is fine.

I just mean that `download_root` should set `download_model(..., cache_dir=download_root)` and not `download_model(..., output_dir=download_root)`.

Comment (jordimas) [reply faster_whisper/transcribe.py]: Let me do a commit.

Is this https://github.com/guillaumekln/faster-whisper/pull/182/commits/48c07efeae9604a5987947a6f57df40e031a52f2 what you suggest?

Comment (guillaumekln) [reply faster_whisper/transcribe.py]: Not exactly.

Here's the full diff I'm suggesting:

```diff
diff --git a/faster_whisper/transcribe.py b/faster_whisper/transcribe.py
index a54b9c6..4d7f0f6 100644
--- a/faster_whisper/transcribe.py
+++ b/faster_whisper/transcribe.py
@@ -112,7 +112,9 @@ class WhisperModel:
             model_path = model_size_or_path
         else:
             model_path = download_model(
-                model_size_or_path, download_root, local_files_only
+                model_size_or_path,
+                local_files_only=local_files_only,
+                cache_dir=download_root,
             )
 
         self.model = ctranslate2.models.Whisper(
diff --git a/faster_whisper/utils.py b/faster_whisper/utils.py
index 649906a..34a310a 100644
--- a/faster_whisper/utils.py
+++ b/faster_whisper/utils.py
@@ -35,6 +35,7 @@ def download_model(
     size: str,
     output_dir: Optional[str] = None,
     local_files_only: Optional[bool] = False,
+    cache_dir: Optional[str] = None,
 ):
     """Downloads a CTranslate2 Whisper model from the Hugging Face Hub.
 
@@ -47,6 +48,7 @@ def download_model(
         the standard Hugging Face cache directory.
       local_files_only:  If True, avoid downloading the file and return the path to the local
         cached file if it exists.
+      cache_dir: Path to the folder where cached files are stored.
 
     Returns:
       The path to the downloaded model.
@@ -66,6 +68,9 @@ def download_model(
         kwargs["local_dir"] = output_dir
         kwargs["local_dir_use_symlinks"] = False
 
+    if cache_dir is not None:
+        kwargs["cache_dir"] = cache_dir
+
     allow_patterns = [
         "config.json",
         "model.bin",
```

**Comment by jordimas** [comment]
Have  a look new @guillaumekln

**Comment by guillaumekln** [comment]
Thanks, I will release this change in a new version.

**Comment by jordimas** [comment]
Thanks for your help on this one!

**Pull request merged**